src/components/charts/LineChart.tsx

```diff
@@ -0,0 +1,185 @@
+import React from "react"
+import merge from "deepmerge"
+
+import {
+  ChartWrapper,
+  RecordSet,
+  RecordEntry,
+  ChartProps,
+} from "./ChartWrapper"
+
+export interface LineChartProps extends ChartProps {
+  strokeWidths?: number | number[]
+  /**
+   * The type of dashed lines for each line. Higher numnbers means more space between dashes.
+   */
+  dashTypes?: number | number[]
+  /**
+   * Whether each line is curved.
+   */
+  curved?: boolean
+  /**
+   * Specify the size of markers for each line. 0 means no markers.
+   */
+  markerSizes?: number | number[]
+  /**
+   * The title of the chart
+   */
+  xMin?: number
+  /**
+   * The maximum value of the x (horizontal) or y axis. Useful when data labels overflow.
+   */
+  xMax?: number
+  /**
+   * The minimum value of the x (horizontal) or y axis.
+   */
+  yMin?: number
+  /**
+   * The maximum value of the x (horizontal) or y axis. Useful when data labels overflow.
+   */
+  yMax?: number
+  /**
+   * The width of the chart object. Defaults to 100% so it is responsive.
+   */
+}
+
+interface DataPoint {
+  x: string | number
+  y: number
+}
+
+interface LineSeriesEntry {
+  name: string
+  data: DataPoint[]
+}
+
+export type LineSeries = LineSeriesEntry[]
+
+export const makeSeries = (
+  data: RecordSet,
+  x: string,
+  y: string[]
+): LineSeries => {
+  const series = y.map(
+    (label: string): LineSeriesEntry => ({
+      name: label,
+      data: data.map(
+        (entry: RecordEntry): DataPoint => ({
+          x: entry[x],
+          y: entry[label] as number,
+        })
+      ),
+    })
+  )
+
+  return series
+}
+
+export const makeDefaultOptions = (
+  strokeWidths: number | number[],
+  dashTypes: number | number[],
+  curved: boolean,
+  markerSizes: number | number[],
+  xMin: number | undefined,
+  xMax: number | undefined,
+  yMin: number | undefined,
+  yMax: number | undefined
+): Record<string, unknown> => ({
+  chart: {
+    animations: {
+      enabled: true,
+      easing: "easeinout",
+      speed: 800,
+      animateGradually: {
+        enabled: true,
+        delay: 30,
```

Comment:
It is generally a good idea to write numbers like `delay: 30,  // seconds`

Reply:
Will do :)